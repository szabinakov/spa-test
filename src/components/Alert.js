import React from "react";
import PropTypes from "prop-types";
import "../styles/Alert.css";

export default function Alert({ message, success }) {
  if (!message) return null;
  return (
    <div className="alertContainer">
      <div className={`alert-${success ? "success" : "error"}`}>{message}</div>
    </div>
  );
}

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};
