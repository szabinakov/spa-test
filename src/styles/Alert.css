.alertContainer {
  display: flex;
  justify-content: center;
}

.alert-success {
  font-weight: bolder;
  color: var(--darkblue);
  background-color: var(--alertgreen);
  margin-top: 30px;
  padding: 5px 20px 5px 20px;
  border-radius: 30px;
}

.alert-error {
  font-weight: bolder;
  color: var(--darkblue);
  background-color: var(--alertred);
  margin-top: 30px;
  padding: 5px 20px 5px 20px;
  border-radius: 30px;
}
