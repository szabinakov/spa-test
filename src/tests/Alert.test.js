import { render } from "@testing-library/react";
import Alert from "../components/Alert.js";

describe("AddUsers", () => {
  const mockMsgGreen = "Mocked Success Message";
  const mockSucc = true;
  const mockMsgRed = "Mocked Error Message";
  const mockUnSucc = false;

  it("renders success message", () => {
    const { asFragment } = render(
      <Alert message={mockMsgGreen} success={mockSucc} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders error message", () => {
    const { asFragment } = render(
      <Alert message={mockMsgRed} success={mockUnSucc} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("it renders every element when success message", () => {
    const { getByTestId } = render(
      <Alert message={mockMsgGreen} success={mockSucc} />
    );

    expect(getByTestId("alert-success")).toHaveClass("alert-success");
  });
  it("it renders every element when error message", () => {
    const { getByTestId } = render(
      <Alert message={mockMsgRed} success={mockUnSucc} />
    );

    expect(getByTestId("alert-error")).toHaveClass("alert-error");
  });
});
