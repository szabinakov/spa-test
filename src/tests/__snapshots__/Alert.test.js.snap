// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUsers renders error message 1`] = `
<DocumentFragment>
  <div
    class="alertContainer"
  >
    <div
      class="alert-error"
      data-testid="alert-error"
    >
      Mocked Error Message
    </div>
  </div>
</DocumentFragment>
`;

exports[`AddUsers renders success message 1`] = `
<DocumentFragment>
  <div
    class="alertContainer"
  >
    <div
      class="alert-success"
      data-testid="alert-success"
    >
      Mocked Success Message
    </div>
  </div>
</DocumentFragment>
`;
